File: cs50/week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h> //has a useful string length function
#include <ctype.h> //has useful character detection functions


int count_letters(string text); //defining the function to be called in main
int count_sentences(string text);
int count_sentences(string text);


int main(void) {

    //prompt the user for a string (a few sentences/sentences from a book)
    string text = get_string("Text: ");

    //calculate and print the number of letters in the prompted string
    int letters = count_letters(text);
    printf("%i letters\n", letters);

    //calculate and print the number of sentences in the prompted string
    int sentences = count_sentences(text);
    printf("%i sentences\n", sentences);

}



int count_letters(string text) {

    //calculate string length
    int length = strlen(text);

    //initialise counter
    int letters = 0;

    //count the number of letters in the string [a to z, lowercase AND uppercase]
    for (int i = 0; i < length; i++) {
        if (isalpha(text[i]))
            letters++;
    }

    return letters;
}



int count_sentences(string text) {

    //calculate string length
    int length = strlen(text);

    //initialise counter
    int sentences = 0;

    //if something has been entered, continue, else, there are no sentences
    if (text[0] != '\0') {
        sentences++; //can assume the text will contain at least 1 word

        //count the number of sentences in the string [any sequence of characters separated by spaces]
        for (int i = 0; i < length; i++) {
            if (isgraph(text[i-1]) && isspace(text[i])) //defined a word as a character followed by a space
                sentences++;
        }
    }

    return sentences;
}



int count_sentences(string text) {

    //calculate string length
    int length = strlen(text);

    //initialise counter
    int sentences = 0;

    //if something has been entered, continue, else, there are no sentences
    if (text[0] != '\0') {
        sentences++; //can assume the text will contain at least 1 sentence

        //count the number of sentences in the string [any sequence of characters separated by spaces]
        for (int i = 0; i < length; i++) {
            if (text[i] == '.' || text[i] == '?' || text[i] == '!') //defined a word as a character followed by a space
                sentences++;
        }
    }

    return sentences;
}

File: cs50/week6/sentimental-readability/readability.py

# program outline

# prompt user for an entry of text
# output the grade level for the text - following Coleman-Liau index
    # count the number of letters
        # can assume a letter is lowercase A to Z, or uppercase A to Z
    # count the number of words
        # any sequence of characters separated by a space dictates the end of a word
    # count the number of sentences
        # can assume any occurence of . ! ? dictates the end of a sentence




def main():
    text = input("Text: ")


    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)


    print(f"Letters: {letters}")
    print(f"Words: {words}")
    print(f"Sentences: {sentences}")



    #spaces = sum(c.isspace() for c in text)
    #numbers = sum(c.isdigit() for c in text)
    #print(f"Spaces: {spaces}")
    #print(f"Numbers: {numbers}")


#def user_input():
    #text = input("Text: ")





# any alphabetical character a-z or A-Z is a letter
def count_letters(text):
    letters = sum(c.isalpha() for c in text)
    return letters


# any sequence of characters separated by a space is a word
def count_words(text):
    words = len(text.split())               # split string into a list of words, count length of list
    return words





# any occurrence of . ! ? indicates the end of a sentence
def count_sentences(text):
    sentences = 0
    return sentences


def grade(text):
    grade = 0
    return grade


main()
